### PR TITLE
more env, server side debug for pipeline ext tests

### PR DESCRIPTION
**Pull request by gabemontero**
@openshift/devex fyi

stems from some recent results with jenkins-client plugin PR testing

### CODE REVIEW

**Comment by gabemontero** [comment]
[testextended][extended:core(openshift pipeline build)]

File: test/extended/builds/pipeline.go [hunk 1]
```diff
@@ -147,7 +147,14 @@ var _ = g.Describe("[builds][Slow] openshift pipeline build", func() {
 
 			// start the build
 			g.By("starting the pipeline build and waiting for it to complete")
-			br, _ := exutil.StartBuildAndWait(oc, "openshift-jee-sample")
+			br, err := exutil.StartBuildAndWait(oc, "openshift-jee-sample")
+			if err != nil || !br.BuildSuccess {
+				exutil.DumpBuildLogs("openshift-jee-sample", oc)
```
Comment (bparees) [inline test/extended/builds/pipeline.go]: since this is a pipeline build, this won't do anything right?

File: test/extended/builds/pipeline.go [hunk 2]
```diff
@@ -176,7 +183,12 @@ var _ = g.Describe("[builds][Slow] openshift pipeline build", func() {
 			o.Expect(err).NotTo(o.HaveOccurred())
 
 			g.By("starting the pipeline build and waiting for it to complete")
-			br, _ := exutil.StartBuildAndWait(oc, "openshift-jee-sample")
+			br, err := exutil.StartBuildAndWait(oc, "openshift-jee-sample")
+			if err != nil || !br.BuildSuccess {
+				exutil.DumpBuildLogs("openshift-jee-sample", oc)
```
Comment (bparees) [inline test/extended/builds/pipeline.go]: same here

File: test/extended/builds/pipeline.go [hunk 3]
```diff
@@ -208,7 +220,12 @@ var _ = g.Describe("[builds][Slow] openshift pipeline build", func() {
...
```
Comment (bparees) [inline test/extended/builds/pipeline.go]: and here

File: test/extended/builds/pipeline.go [hunk 4]
```diff
@@ -240,7 +257,12 @@ var _ = g.Describe("[builds][Slow] openshift pipeline build", func() {
...
```
Comment (bparees) [inline test/extended/builds/pipeline.go]: and here

File: test/extended/builds/pipeline.go [hunk 5]
```diff
@@ -269,8 +291,15 @@ var _ = g.Describe("[builds][Slow] openshift pipeline build", func() {
...
 			g.By("starting the pipeline build and waiting for it to complete")
-			br, _ := exutil.StartBuildAndWait(oc, "sample-pipeline-openshift-client-plugin")
+			br, err := exutil.StartBuildAndWait(oc, "sample-pipeline-openshift-client-plugin")
 			debugAnyJenkinsFailure(br, oc.Namespace()+"-sample-pipeline-openshift-client-plugin", oc, true)
+			if err != nil || !br.BuildSuccess {
+				exutil.DumpBuildLogs("ruby", oc)
+				exutil.DumpDeploymentLogs("mongodb", oc)
+				exutil.DumpDeploymentLogs("jenkins-second-deployment", oc)
+				exutil.ExaminePodDiskUsage(oc)
+				exutil.ExamineDiskUsage()
```
Comment (bparees) [inline test/extended/builds/pipeline.go]: DumpBuildLogs appears to already call ExaminePodDiskUsage and ExamineDiskUsage

**Comment by bparees** [comment]
two main points:

1) buildlogs won't work on pipeline builds
2) dumpbuildlogs invokes ExaminediskUsage and ExaminePodDiskUsage already, so in blocks where you are dumping build logs, you don't need to explicilty call those other 2.

**Comment by gabemontero** [comment]
corrections pushed (last test job did not encounter errors that would invoke new diagnostics, though I still think our *plugin* PR tests are more susceptible somehow)

**Comment by bparees** [comment]
lgtm and extended tests passed, will manually merge on passing [test]

**Comment by bparees** [comment]
unless [merge] gets to it first.

**Comment by gabemontero** [comment]
11th on the merge queue as of this posting ... manual merge on passing test will win the race

**Comment by gabemontero** [comment]
conformance gce had a failure in the `PROVISION TEST CLUSTER` stage.  Seems similar to flake https://github.com/openshift/origin/issues/14575 opened by @bparees but not exact.

The first inkling of some trouble was:

```
TASK [openshift_node : Start and enable node] **********************************
Thursday 29 June 2017  16:01:54 +0000 (0:00:00.156)       0:15:03.087 ********* 
FAILED - RETRYING: Start and enable node (1 retries left).
FAILED - RETRYING: Start and enable node (1 retries left).
FAILED - RETRYING: Start and enable node (1 retries left).
FAILED - RETRYING: Start and enable node (1 retries left).
ok: [ci-prtest-5a37c28-3889-ig-m-k63s]
ok: [ci-prtest-5a37c28-3889-ig-n-2l38]
ok: [ci-prtest-5a37c28-3889-ig-n-0xsr]
ok: [ci-prtest-5a37c28-3889-ig-n-j2s8]
```

but perhaps it recovered given the final set of `ok` messages.

But then we got this:

```
RUNNING HANDLER [openshift_node : restart node] ********************************
Thursday 29 June 2017  16:02:40 +0000 (0:00:00.053)       0:15:49.265 ********* 
fatal: [ci-prtest-5a37c28-3889-ig-m-k63s]: FAILED! => {"changed": false, "failed": true, "msg": "Unable to restart service origin-node: Job for origin-node.service failed because the control process exited with error code. See \"systemctl status origin-node.service\" and \"journalctl -xe\" for details.\n"}
fatal: [ci-prtest-5a37c28-3889-ig-n-j2s8]: FAILED! => {"changed": false, "failed": true, "msg": "Unable to restart service origin-node: Job for origin-node.service failed because the control process exited with error code. See \"systemctl status origin-node.service\" and \"journalctl -xe\" for details.\n"}
fatal: [ci-prtest-5a37c28-3889-ig-n-0xsr]: FAILED! => {"changed": false, "failed": true, "msg": "Unable to restart service origin-node: Job for origin-node.service failed because the control process exited with error code. See \"systemctl status origin-node.service\" and \"journalctl -xe\" for details.\n"}
fatal: [ci-prtest-5a37c28-3889-ig-n-2l38]: FAILED! => {"changed": false, "failed": true, "msg": "Unable to restart service origin-node: Job for origin-node.service failed because the control process exited with error code. See \"systemctl status origin-node.service\" and \"journalctl -xe\" for details.\n"}
```

And 

```
Failure summary:

  1. Host:     ci-prtest-5a37c28-3889-ig-m-k63s
     Play:     Configure nodes
     Task:     openshift_node : restart node
     Message:  Unable to restart service origin-node: Job for origin-node.service failed because the control process exited with error code. See "systemctl status origin-node.service" and "journalctl -xe" for details.
               

  2. Host:     ci-prtest-5a37c28-3889-ig-n-j2s8
     Play:     Configure nodes
     Task:     openshift_node : restart node
     Message:  Unable to restart service origin-node: Job for origin-node.service failed because the control process exited with error code. See "systemctl status origin-node.service" and "journalctl -xe" for details.
               

  3. Host:     ci-prtest-5a37c28-3889-ig-n-0xsr
     Play:     Configure nodes
     Task:     openshift_node : restart node
     Message:  Unable to restart service origin-node: Job for origin-node.service failed because the control process exited with error code. See "systemctl status origin-node.service" and "journalctl -xe" for details.
               

  4. Host:     ci-prtest-5a37c28-3889-ig-n-2l38
     Play:     Configure nodes
     Task:     openshift_node : restart node
     Message:  Unable to restart service origin-node: Job for origin-node.service failed because the control process exited with error code. See "systemctl status origin-node.service" and "journalctl -xe" for details.
```

@stevekuznetsov @smarterclayton @bparees - should new flake be created or reuse https://github.com/openshift/origin/issues/14575 ?

**Comment by bparees** [comment]
@gabemontero open a new one, we'll let @stevekuznetsov and @smarterclayton close if it they feel it's a dupe of #14575 

meanwhile: [test]

**Comment by gabemontero** [comment]
created flake https://github.com/openshift/origin/issues/14964

**Comment by smarterclayton** [comment]
It's not, it's a separate issue that has a bugzilla already with networking
and host namespace.

**Comment by smarterclayton** [comment]
It's https://bugzilla.redhat.com/show_bug.cgi?id=1465361

**Comment by bparees** [comment]
failed in GCE de-provision, extended-test-only changes(which passed), manually merging.

**Comment by gabemontero** [comment]
Yeah there is a dense string that is dumped on the deprovision failure.

I found: 

```
"ERROR: (gcloud.compute.instance-groups.managed.delete) Some requests did not succeed:", " - Internal Error""
```

In the middle of quite a few successful `Deleted` records.

Couldn't parse anythings else of relevance.

thanks for the manual merge @bparees

**Pull request merged**